ens/main.py

```diff
@@ -335,16 +395,24 @@ def set_text(
 
         r = self.resolver(normal_name)
         if r:
-            return r.functions.setText(node, key, value).transact(transaction_dict)
+            if _resolver_supports_interface(r, GET_TEXT_INTERFACE_ID):
+                return r.functions.setText(node, key, value).transact(transaction_dict)
+            else:
+                raise UnsupportedFunction(
+                    f"Resolver for name `{name}` does not support `text` function"
+                )
         else:
-            raise UnownedName("claim domain using setup_address() first")
+            raise ResolverNotFound(
+                f"No resolver found for name `{name}`. It is likely the name contains an "
+                "unsupported top level domain (tld)."
```

Comment:
I think that any domain with an `eth` tld should at least return the resolver that is set on `eth` unless the subdomain has been claimed and a resolver is set there, then it would return that resolver.

So I think this should always be a bad `tld` unless for some reason `eth` parent doesn't have a resolver set which shouldn't happen. Unless I'm missing an edge case somewhere?